hutch_python/ipython_session_timer.py

```diff
@@ -36,41 +36,62 @@ class IPythonSessionTimer:
 
     Attributes
     ----------
+    curr_time: float
+        The current time in seconds.
+
     max_idle_time: int
         The maximum number of seconds a user session can be idle (currently set
         to 172800 seconds or 48 hours).
+
+    last_active_time: float
+        The time of the last user activity in this session.
+
+    idle_time: float
+        The amount of time the user session has been idle.
     '''
 
     def __init__(self, ipython):
         global max_idle_time
-        self.max_idle_time = max_idle_time
+        self.curr_time = 0
+        self.max_idle_time = 30  # max_idle_time
+        self.last_active_time = 0
+        self.idle_time = 0
         self.user_active = False
 
-        # _set_last_active_time() function will trigger every time user runs a cell
         ipython.events.register('pre_run_cell', self._set_user_active)
         ipython.events.register('post_run_cell', self._set_user_inactive)
 
-    def _set_user_active(self, result):
+    def _set_user_active(self):
         self.user_active = True
+        self.last_active_time = time.time()
```

Comment:
funny python-y thing.  We should use `time.monotonic()` here, `time.time()` is the system time I believe, and can change.